components/script/dom/globalscope.rs

```diff
@@ -3431,15 +3431,30 @@ impl GlobalScope {
         if Some(false) == self.inherited_secure_context {
             return false;
         }
-        if let Some(top_level_creation_url) = self.top_level_creation_url() {
-            if top_level_creation_url.scheme() == "blob" &&
-                Some(true) == self.inherited_secure_context
-            {
-                return true;
-            }
-            return top_level_creation_url.is_potentially_trustworthy();
+        // Step 1. If environment is an environment settings object, then:
+        // Step 1.1. Let global be environment's global object.
+        match self.top_level_creation_url() {
+            None => {
+                // Workers and worklets don't have a top-level creation URL
+                assert!(
+                    self.downcast::<WorkerGlobalScope>().is_some() ||
```

Comment:
Step 1.2.1, which is for `WorkerGlobalScope`, isn't implemented here. Let's add a TODO.

> If global is a [WorkerGlobalScope](https://html.spec.whatwg.org/multipage/workers.html#workerglobalscope), then:
>

> 1. If global's [owner set](https://html.spec.whatwg.org/multipage/workers.html#concept-WorkerGlobalScope-owner-set)[0]'s [relevant settings object](https://html.spec.whatwg.org/multipage/webappapis.html#relevant-settings-object) is a [secure context](https://html.spec.whatwg.org/multipage/webappapis.html#secure-context), then return true.

> 2. Return false.